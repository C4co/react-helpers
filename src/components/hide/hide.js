import React from "react"
import PropTypes from "prop-types"

export function Hide(props) {
  if (props.if) {
    return null
  }

  return <> {props.children} </>
}

Hide.propTypes = {
  /** Condition to show or hide the component. */
  if: PropTypes.bool,

  /** Children content */
  children: PropTypes.any
}

Hide.defaultProps = {
  if: true
}
