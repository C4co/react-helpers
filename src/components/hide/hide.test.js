import { Hide } from "./hide"
import { mount } from "enzyme"
import React from "react"

describe("<Hide> component", () => {
  test("Check hidden", () => {
    const hidden = mount(
      <Hide if>
        <h1>i am hidden</h1>
      </Hide>
    )

    expect(hidden.contains(<h1>i am hidden</h1>)).toBeFalsy()
    expect(hidden).toMatchSnapshot()
  })

  test("Check visibility", () => {
    const visible = mount(
      <Hide if={false}>
        <h1>i am visible</h1>
      </Hide>
    )

    expect(visible.contains(<h1>i am visible</h1>)).toBeTruthy()
    expect(visible).toMatchSnapshot()
  })
})
