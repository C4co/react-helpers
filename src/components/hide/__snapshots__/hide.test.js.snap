// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hide> component Check hidden 1`] = `
<Hide
  if={true}
/>
`;

exports[`<Hide> component Check visibility 1`] = `
<Hide
  if={false}
>
   
  <h1>
    i am visible
  </h1>
   
</Hide>
`;
